### PR TITLE
Update AppWrappers to v0.21.1

### PR DESCRIPTION
**Pull request by dgrove-oss**
Highlights:
  * Update to Kueue 0.7.1
  * Remove ChildAdmissionController only needed by Kueue 0.6
  * Flexible configuration of Kueue's GenericJob Reconciller

Full Changelog:
  https://github.com/project-codeflare/appwrapper/compare/v0.20.2...v0.21.0

### CODE REVIEW

**Review by KPostOffice — COMMENTED** [review]
This all look sane to me. Do you have any clue about the failing e2e test. Could it be due to some change in logic in the AW controller?

**Comment by dgrove-oss** [comment]
At first glance in the logs, it looks like the AppWrapper+RayCluster workload is pending because a GPU isn't available, but I haven't analyzed fully to figure out if that is just a transitory problem that might resolve once the concurrently running test that is using the GPU first finishes.. 
```
10T14:54:39.771940512Z","logger":"scheduler","caller":"scheduler/scheduler.go:263","msg":"Workload requires preemption, but there are no candidate workloads allowed for preemption","workload":{"name":"appwrapper-raycluster-af4b6","namespace":"test-ns-vpd2b"},"clusterQueue":{"name":"e2e-cluster-queue"},"preemption":{"reclaimWithinCohort":"Never","borrowWithinCohort":{"policy":"Never"},"withinClusterQueue":"Never"}}
{"level":"Level(-2)","ts":"2024-07-10T14:54:39.771986932Z","logger":"scheduler","caller":"scheduler/scheduler.go:619","msg":"Workload re-queued","workload":{"name":"appwrapper-raycluster-af4b6","namespace":"test-ns-vpd2b"},"clusterQueue":{"name":"e2e-cluster-queue"},"queue":{"name":"lq-vbsxg","namespace":"test-ns-vpd2b"},"requeueReason":"","added":true}
{"level":"Level(-2)","ts":"2024-07-10T14:54:39.772046413Z","logger":"cluster-queue-reconciler","caller":"core/clusterqueue_controller.go:330","msg":"Got generic event","obj":{"name":"appwrapper-raycluster-af4b6","namespace":"test-ns-vpd2b"},"kind":"/, Kind="}
{"level":"debug","ts":"2024-07-10T14:54:39.780801696Z","logger":"events","caller":"recorder/recorder.go:104","msg":"couldn't assign flavors to pod set raycluster-0-1: insufficient unused quota for nvidia.com/gpu in flavor default-flavor, 1 more needed","type":"Normal","object":{"kind":"Workload","namespace":"test-ns-vpd2b","name":"appwrapper-raycluster-af4b6","uid":"fd964893-f987-47b1-9858-7001fa2c6a89","apiVersion":"kueue.x-k8s.io/v1beta1","resourceVersion":"3691"},"reason":"Pending"}
```

**Comment by sutaakar** [comment]
I have already seen this intermittently, seems caused by AppWrapper workload not being admitted:
```
2024-07-10 14:54:39  | 2024-07-10 14:54:39  | appwrapper-raycluster-af4b6.17e0e1c1b799975a  | -          | Normal  | Pending          | couldn't assign flavors to pod set raycluster-0-1: insufficient unused quota for nvidia.com/gpu in flavor default-flavor, 1 more needed  | 
2024-07-10 14:54:39  | 2024-07-10 14:54:39  | raycluster.17e0e1c1b7086150                   | -          | Normal  | CreatedWorkload  | Created Workload: test-ns-vpd2b/appwrapper-raycluster-af4b6      
```

Though I wasn't able to reproduce it yet.
Will try to get some more context to that failure (LocalQueue status, Workloads and such).

**Comment by dgrove-oss** [comment]
Going to hold because back leveling Go to 1.22.2 is  more trouble that it is worth.

**Comment by sutaakar** [comment]
e2e test instability is addressed in https://github.com/project-codeflare/codeflare-operator/pull/591

File: go.mod [hunk 1]
```diff
@@ -34,13 +34,18 @@ replace go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp => go.open
 
 replace github.com/jackc/pgx/v4 => github.com/jackc/pgx/v5 v5.5.4
 
+// These replace directives support the backlevel go version required by ODH build
+replace github.com/project-codeflare/appwrapper v0.21.1 => github.com/project-codeflare/appwrapper v0.21.2-0.20240712173553-5b007c947b37
+
+replace sigs.k8s.io/kueue v0.7.1 => github.com/opendatahub-io/kueue v0.7.0-odh-test
+
```
Comment (dgrove-oss) [inline go.mod:41]: Not thrilled with these replace directives, but we can't have the go.version at 1.22.2 and depend directly on the upstream kueue and appwrapper releases which are at 1.22.4 without breaking all the go tooling.

**Review by sutaakar — APPROVED** [review]

**Pull request merged**